### ISSUE TITLE
strcpy-param-overlap in proxy.c

### ISSUE THREAD

**Issue by lenix123** [3proxy/3proxy#1108]
Hi, we were fuzzing 3proxy and have found AddressSanitizer fail with `strcpy-param-overlap`.
Steps to reproduce (we use ):
1. Get the 3proxy 0.9.3 release:
``` bash
wget https://github.com/z3APA3A/3proxy/archive/0.9.3.tar.gz && tar xf 0.9.3.tar.gz && rm 0.9.3.tar.gz
``` 
2. Compile the program with AddressSanitizer. For that you can add flag `-fsanitize=address` to CFLAGS and LDFLAGS in Makefile. We compiled with clang-12 on ubuntu:20.04
``` bash
cd 3proxy-0.9.3 && ln -s Makefile.Linux Makefile
make
```
3. Run http proxy:
``` bash
./bin/proxy
```
4. In another terminal send the `crash` from attachment to the proxy with `nc`:
``` bash
nc 127.0.0.1 3128 < crash.txt
Ctrl+C
```
[crash.txt](https://github.com/user-attachments/files/18207585/crash.txt)  

Found by Leonid Reviakin with AFL++

**Comment by z3APA3A** [comment]
` % nc 127.0.0.1 3128 < crash.txt
HTTP/1.0 502 Bad Gateway
Connection: close
Content-type: text/html; charset=utf-8

<html><head><title>502 Bad Gateway</title></head>
<body><h2>502 Bad Gateway</h2><h3>Host Not Found or connection failed</h3></body></html>
`

Can't reproduce it in current version. Latest released stable version is 0.9.4, 0.9.3 is long time outdated.

**Comment by z3APA3A** [comment]
Replaced strcpy with memmove anyway in 6387bed4f282c43ed1fcca715a73fabbecb0390b
